idx/memory/tag_query.go

```diff
@@ -14,10 +18,13 @@ var (
 )
 
 const (
-	EQUAL = iota
-	NOT_EQUAL
-	MATCH
-	NOT_MATCH
+	EQUAL      = iota // =
+	NOT_EQUAL         // !=
+	MATCH             // =~        regular expression
+	MATCH_TAG         // __tag=~   relies on special key __tag
+	NOT_MATCH         // !=~
+	PREFIX            // ^=        exact prefix, not regex
+	PREFIX_TAG        // __tag^=   exact prefix with tag
```

Comment:
are all these exactly the same as in graphite? we should document somewhere which operators we support and what the various tricks are (e.g. `tag!=` doing a check that the tag is non zero afaik). this is probably not the right place. maybe docs/http-api.md is, or we can also just refer to the graphite docs if we're 100% compatible.
we also currently don't really document the kind of regular patterns (non-tag) a user can use. not sure if we want to maintain all that ourself though, i would rather justpoint to graphite docs

Reply:
i think this should be the documentation used, because it's the reference implementation as well: http://graphite.readthedocs.io/en/latest/tags.html

Reply:
not all those operations are supported by graphite. some of them were only implemented because we need them internally to satisfy queries like for example by tag prefix (for tag autocomplete). if a user figures out how to use them, that's fine, but they are not intended to be used by users and hence not documented in the reference documentation ^^

Reply:
ok so in tag_query.go lets link to that URL and also clearly mark the non-standard ones.
looking at docs/http-api.md again, it's not the right place because it just talks about the different user accessible http endpoints, such as /metrics/find and /render; but doesn't go into much more detail regarding graphite processing functions

Reply:
a50882b